JPetMC.cpp

```diff
@@ -22,31 +22,33 @@
 #include <G4INCLRandom.hh>
 #include <G4UImanager.hh>
 #include "Info/EventMessenger.h"
-#include <fstream> 
+#include <fstream>
 #include <random>
+#include <TRandom3.h>
 
 void setRandomSeed()
 {
-  long seeds[2];
-  
+  long seed;
+
   std::uniform_int_distribution<long> d(0, LONG_MAX);
   std::random_device rd1;
 
-  seeds[0] = d(rd1);
-  seeds[1] = d(rd1);
-  G4Random::setTheSeeds(seeds);
+  seed = d(rd1);
+
+  G4Random::setTheEngine(new CLHEP::MTwistEngine());
+  G4Random::setTheSeed(seed);
 
-  if (EventMessenger::GetEventMessenger()->SaveSeed()) {
-    std::ofstream file;
-    file.open ("seed", std::ofstream::out | std::ofstream::app);
-    file << seeds[0] << "\n" << seeds[1] << "\n";
-    file.close();
-  }
 }
 
 
 int main (int argc, char** argv)
 {
+  delete gRandom;
+  gRandom = new TRandom3(0);
+
+  std::cout << gRandom ->GetSeed() << std::endl;
```

Comment:
Remove the cout. Anyway, the above lines look like leftovers from some quick test, shouldn't they be removed?

Reply:
Cout is a leftover. Lines 46 and 47 are crucial as explained previously.